src/network/msgpackobject.cc

```diff
@@ -157,74 +181,92 @@ ObjectType MsgpackObject::type() const {
 }
 
 bool& MsgpackObject::getBool() {
+  if (obj_type != ObjectType::BOOLEAN) throw proto::SchemaError("Wrong MsgpackObject type when trying to get bool");
   return value.boolean;
 }
 
-bool MsgpackObject::getBool() const{
+bool MsgpackObject::getBool() const {
+  if (obj_type != ObjectType::BOOLEAN) throw proto::SchemaError("Wrong MsgpackObject type when trying to get bool");
   return value.boolean;
 }
 
 uint64_t& MsgpackObject::getUnsignedInt() {
+  validateGetUnsignedInt();
```

Comment:
Nope.  To robi type-punning na polach unii, co nie jest dozwolone.  Trzeba czytać sint albo uint zgodnie z prawdziwym typem.

To znaczy też, że nie możemy napisać getUnsignedInt zwracającego refa - IMO najlepiej byłoby w ogóle naprawić wszystkie get* zwracające ref - to brzmi jak zadanie dla setterów